src/main/scala-2.12/org/apache/james/gatling/imap/scenari/ImapListMessagesBodyStructureScenario.scala

```diff
@@ -7,19 +7,21 @@ import com.linagora.gatling.imap.protocol.command.MessageRanges
 import io.gatling.core.Predef._
 import io.gatling.core.feeder.FeederBuilder
 import io.gatling.core.structure.ScenarioBuilder
+import org.apache.james.gatling.jmap.MailboxName
 
 import scala.concurrent.duration._
 
 class ImapListMessagesBodyStructureScenario {
-
-  val listMessagesBodyStructure = exec(imap("select").select("INBOX").check(ok))
+  val listMessagesBodyStructure = exec(imap("select").select("${mailboxName}").check(ok))
     .exec(imap("fetch").fetch(MessageRanges(From(1)), AttributeList("BODY")).check(ok))
 
-  def generate(feeder: FeederBuilder): ScenarioBuilder =
+  def generate(feeder: FeederBuilder, mailboxes: List[MailboxName]): ScenarioBuilder =
     scenario("ImapListMessagesBodyStructureScenario")
       .feed(feeder)
       .pause(1.second)
       .exec(imap("Connect").connect()).exitHereIfFailed
       .exec(imap("login").login("${username}", "${password}").check(ok))
-      .exec(listMessagesBodyStructure)
+      .exec(foreach(mailboxes.map(mailboxName => mailboxName.name), "mailboxName") {
```

Comment:
I don't think that's what we expect from the scenario. I would say we select a random folder.